code/src/reducers/gameInfo.js

```diff
@@ -31,7 +35,12 @@ export const getMoves = (action) => {
         direction: action.direction
       })
     }).then(res => res.json())
-      .then(data => 
-        dispatch(gameDetails.actions.setMoves(data)));
+      .then(data => {
+        if (data.actions.length === 0) {
+          dispatch(gameDetails.actions.gameFinished())
+        }
+        dispatch(gameDetails.actions.setMoves(data));
+    });
       };
 };
```

Comment:
Interesting to see other ways of implementing the 'game over' function, especially with an if statement and a length property! We have done it a bit different, by clearing the local storage-nonetheless, this is once again smart and simple. ☺️